main.go

```diff
@@ -914,7 +914,27 @@ func transformCompile(args []string) ([]string, error) {
 		return nil, err
 	}
 
-	tf.findReflectFunctions(files)
+	prog := ssa.NewProgram(fset, 0)
+
+	// Create SSA packages for all imports.
+	// Order is not significant.
+	created := make(map[*types.Package]bool)
+	var createAll func(pkgs []*types.Package)
+	createAll = func(pkgs []*types.Package) {
+		for _, p := range pkgs {
+			if !created[p] {
+				created[p] = true
+				prog.CreatePackage(p, nil, nil, true)
+				createAll(p.Imports())
```

Comment:
do we need to recurse? the docs for Package.Build say:

> Precondition: CreatePackage must have been called for all of p's direct imports (and hence its direct imports must have been error-free). 

so I don't think we need to call CreatePackage for indirect imports.

Reply:
This has been adapted from [ssautil.BuildPackage](https://cs.opensource.google/go/x/tools/+/refs/tags/v0.5.0:go/ssa/ssautil/load.go;l=133). I know that at least all direct imports are needed but I didn't actually check if indirect ones are required.

Reply:
Tested, yes we actually need to recurse `ssaPkg.Build()` panics otherwise

Reply:
ack, thanks for confirming